src/EditorFeatures/Text/Extensions.SnapshotSourceText.cs

```diff
@@ -334,25 +338,19 @@ public override IReadOnlyList<TextChangeRange> GetChangeRanges(SourceText oldTex
                     }
                 }
 
-                var oldSnapshot = oldText.FindCorrespondingEditorTextSnapshot();
-                var newSnapshot = this.FindCorrespondingEditorTextSnapshot();
+                var oldSnapshot = oldText.FindCorrespondingEditorTextImage();
+                var newSnapshot = this.FindCorrespondingEditorTextImage();
                 return GetChangeRanges(oldSnapshot, oldText.Length, newSnapshot);
             }
 
-            private IReadOnlyList<TextChangeRange> GetChangeRanges(ITextSnapshot oldSnapshot, int oldTextLength, ITextSnapshot newSnapshot)
+            private IReadOnlyList<TextChangeRange> GetChangeRanges(ITextImage oldSnapshot, int oldTextLength, ITextImage newSnapshot)
             {
                 if (oldSnapshot == null ||
-                    newSnapshot == null ||
-                    oldSnapshot.TextBuffer != newSnapshot.TextBuffer)
```

Comment:
ITextImage has an Identifier property that corresponds to the source text buffer.
Two images from different snapshots on the same buffer will have the same identifier